gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapper.java

```diff
@@ -0,0 +1,252 @@
+import uk.nhs.adaptors.pss.translator.util.EhrResourceExtractorUtil;
+import uk.nhs.adaptors.pss.translator.util.ParticipantReferenceUtil;
+
+@Service
+@AllArgsConstructor
+public class ImmunizationMapper {
+    private static final String META_PROFILE = "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Immunization-1";
+    private static final String IMMUNIZATION_SNOMED_CODE = "2.16.840.1.113883.2.1.3.2.3.15";
+    private static final String IDENTIFIER_SYSTEM = "https://PSSAdaptor/";
+    private static final String VACCINE_PROCEDURE_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect"
+        + "-VaccinationProcedure-1";
+    private static final String END_DATE_PREFIX = "End Date: ";
+    private static final String VACCINATION_CODING_EXTENSION_URL = "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-coding"
+        + "-sctdescid";
+    private static final String RECORDED_DATE_EXTENSION_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-DateRecorded-1";
+
+    private CodeableConceptMapper codeableConceptMapper;
+
+    public List<Immunization> mapToImmunization(RCMRMT030101UK04EhrExtract ehrExtract, Patient patientResource,
+        List<Encounter> encounterList) {
+        List<Immunization> mappedImmunizationResources = new ArrayList<>();
+        var ehrCompositionList = EhrResourceExtractorUtil.extractValidImmunizationEhrCompositions(ehrExtract);
+
+        ehrCompositionList.forEach(ehrComposition -> {
+            var immunizationObservationStatements = getImmunizationObservationStatements(ehrComposition);
+
+            immunizationObservationStatements
+                .forEach(observationStatement -> {
+                    var mappedImmunization = mapImmunization(ehrComposition, observationStatement, patientResource,
+                        encounterList);
+                    mappedImmunizationResources.add(mappedImmunization);
+                });
+        });
+
+        return mappedImmunizationResources;
+    }
+
+    private Immunization mapImmunization(RCMRMT030101UK04EhrComposition ehrComposition,
+        RCMRMT030101UK04ObservationStatement observationStatement, Patient patientResource, List<Encounter> encounterList) {
+        Immunization immunization = new Immunization();
+
+        var id = observationStatement.getId().getRoot();
+        var identifier = getIdentifier(id);
+        var date = getObservationDate(observationStatement);
+
+        var practitioner = ParticipantReferenceUtil.getParticipantReference(observationStatement.getParticipant(), ehrComposition);
+        var encounter = getEncounterReference(encounterList, ehrComposition.getId());
+
+        immunization.getMeta().addProfile(META_PROFILE);
+        immunization.getIdentifier().add(identifier);
+        immunization.getExtension().add(createVaccineProcedureExtension(observationStatement));
+        immunization.getExtension().add(createRecordedTimeExtension(ehrComposition));
```

Comment:
You can do this instead
`immunization.addIdentifier(identifier)
immunization.addExtension()`